### PR TITLE
Make optimizer reporter configurable in sbt again

### PR DESCRIPTION
**Pull request by pbatko**

### CODE REVIEW

File: sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala [hunk 1]
```diff
@@ -25,6 +25,9 @@ import scalanative.sbtplugin.testinterface.ScalaNativeFramework
 
 object ScalaNativePluginInternal {
 
+  val optimizerReporterOutPath =
```
Comment (pbatko) [inline sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Is it ok to have a key for the reporter back if it's in `ScalaNativePluginInternal`? @densh @Duhemm

Comment (ekrich) [reply sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Does it make sense to have a build relative default directory?

Comment (ekrich) [reply sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Well, we could have a switch to turn it on and off and have a project relative destination in the target directory. I think that would work for each project that uses the plugin. @Duhemm What do you think?

**Comment by densh** [comment]
Optimizer reporter is a useful concept but currently it doesn't exist any more due to a complete optimizer rewrite. We need to redesign this feature from scratch as requested again in #1414.

**Pull request closed without merging**